benchmark/benchmain/main.go

```diff
@@ -244,6 +247,9 @@ func unconstrainedStreamBenchmark(start startFunc, stop ucStopFunc, bf stats.Fea
 				}
 				sender(pos)
 				atomic.AddUint64(&req, 1)
+				if maxSleep > 0 {
+					time.Sleep(time.Duration(math_rand.Intn(maxSleep)))
+				}
```

Comment:
Note that delay must be random. I first tried using a fixed delay and my optimization actually showed regression instead of improvement. The reason was that all concurrent goroutines send their requests exactly at the same time, and in such case the fact that we can share buffers between connections doesn't help as all connections are utilized simultaneously. After I modified the delay to be random I saw a big improvement in memory consumption.